include/seqan/blast/blast_base.h

```diff
@@ -0,0 +1,669 @@
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+// Author: Hannes Hauswedell <hannes.hauswedell@fu-berlin.de>
+// ==========================================================================
+// Module for handling NCBI Blast I/O and E-Value computation
+// ==========================================================================
+
+#ifndef SEQAN_EXTRAS_BLAST_BLAST_BASE_H_
+#define SEQAN_EXTRAS_BLAST_BLAST_BASE_H_
+
+namespace seqan {
+
+// ============================================================================
+// Forwards
+// ============================================================================
+
+// ============================================================================
+// Tags, Classes, Enums
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Enum BlastProgram
+// ----------------------------------------------------------------------------
+
+/*!
+ * @enum BlastProgram
+ * @brief Enum with BLAST program spec
+ * @signature enum class BlastProgram : uint8_t { ... };
+ *
+ * @headerfile <seqan/blast.h>
+ * @see BlastProgramTag
+ *
+ * @val BlastProgram BlastProgram::BLASTN
+ * @brief Nucleotide Query VS Nucleotide Subject
+ *
+ * @val BlastProgram BlastProgram::BLASTP
+ * @brief Protein Query VS Protein Subject
+ *
+ * @val BlastProgram BlastProgram::BLASTX
+ * @brief translated Nucleotide Query VS Protein Subject
+ *
+ * @val BlastProgram BlastProgram::TBLASTN
+ * @brief Protein Query VS translated Nucleotide Subject
+ *
+ * @val BlastProgram BlastProgram::TBLASTX
+ * @brief translated Nucleotide Query VS translated Nucleotide Subject
+ *
+ * @val BlastProgram BlastProgram::UNKNOWN
+ * @brief Unkown type. Used internally and to signify that the type could not be inferred from the file.
+ */
+enum class BlastProgram : uint8_t
```

Comment:
As I previously said: before there was one BlastFormat which had multiple template arguments for the fileformate (Tabular, TabularWithHeader, Report, poossibly XML...), the program type (BlastN, BlastP...), the generation (blastall vs blast+). You said that it was super confusing and that it needed to change.
Then I treied it with the TagSelector but obiously this then results in huge confusing number of possible combinations that all get a tag... so instead I want with the above :)